apps/container/src/components/SearchBar.vue

```diff
@@ -0,0 +1,172 @@
+<template>
+  <FeedbackAlert v-if="isErrorTeachers" text="Erro ao buscar professores" />
+  <FeedbackAlert v-if="isErrorSubjects" text="Erro ao buscar disciplinas" />
+  <div class="wrapper w-100 mb-5">
+    <v-text-field
+      variant="solo"
+      v-model="searchTerm"
+      placeholder="Digite o nome do professor ou disciplina"
+      @input="search"
+      class="mb-1"
+      hide-details
+      :prepend-inner-icon="
+        isFetchingTeachers || isFetchingSubjects
+          ? 'mdi-loading mdi-spin'
+          : 'mdi-magnify'
+      "
+      clearable
+      @click:clear="clear"
+    >
+    </v-text-field>
+    <v-list
+      v-if="
+        processedResults.length && (router.currentRoute.value.query.q as string)
```

Comment:
por ser um v-if, não precisa desse `as`
```suggestion
        processedResults.length && router.currentRoute.value.query.q
```